File: src/main/java/com/wgzhao/addax/admin/model/EtlTable.java
package com.wgzhao.addax.admin.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Formula;

import java.util.Date;

/**
 * TB_IMP_ETL 实体类
 *
 * @author
 */
@Entity
@Table(name = "etl_table")
@Setter
@Getter
@Data
public class EtlTable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "source_db", length = 32, nullable = false)
    private String sourceDb;

    @Column(name = "source_table", length = 64, nullable = false)
    private String sourceTable;

    @Column(name = "target_db", length = 50, nullable = false)
    private String targetDb;

    @Column(name = "target_table", length = 200, nullable = false)
    private String targetTable;

    @Column(name = "part_kind", length = 1)
    private String partKind;

    @Column(name = "part_name", length = 20)
    private String partName;

    @Column(name = "filter", length = 2000, nullable = false)
    private String filter;

    @Column(name = "status", length = 1)
    private String status;

    @Column(name = "kind", length = 1)
    private String kind;

    @Column(name = "update_flag", length = 1)
    private String updateFlag;

    @Column(name = "create_flag", length = 1)
    private String createFlag;

    @Column(name = "retry_cnt")
    private Integer retryCnt;

    @Column(name = "start_time")
    private Date startTime;

    @Column(name = "end_time")
    private Date endTime;

    @Column(name = "max_runtime")
    private Integer maxRuntime;

    @ManyToOne
    @JoinColumn(name = "sid", referencedColumnName = "id")
    private EtlSource etlSource;

    @Column(name = "duration", nullable = false)
    private Long duration;

    @Formula("LOWER(source_db || '.' || source_table || target_db || '.' || target_table || part_kind || part_name || filter)")
    private String filterColumn;
}

File: src/main/kotlin/com/wgzhao/addax/admin/model/EtlTable.kt
package com.wgzhao.addax.admin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import org.hibernate.annotations.Formula
import java.util.*

/**
 * TB_IMP_ETL 实体类
 */
@Entity
@Table(name = "etl_table")
data class EtlTable(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long? = null,

    @Column(name = "source_db", length = 32, nullable = false)
    var sourceDb: String? = null,

    @Column(name = "source_table", length = 64, nullable = false)
    var sourceTable: String? = null,

    @Column(name = "target_db", length = 50, nullable = false)
    var targetDb: String? = null,

    @Column(name = "target_table", length = 200, nullable = false)
    var targetTable: String? = null,

    @Column(name = "part_kind", length = 1)
    var partKind: String? = null,

    @Column(name = "part_name", length = 20)
    var partName: String? = null,

    @Column(name = "part_format", length = 10)
    var partFormat: String? = null,

    @Column(name = "filter", length = 2000, nullable = false)
    var filter: String? = null,

    @Column(name = "status", length = 1)
    var status: String? = null,

    @Column(name = "kind", length = 1)
    var kind: String? = null,

    @Column(name = "retry_cnt")
    var retryCnt: Int? = null,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Column(name = "start_time")
    var startTime: Date? = null,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Column(name = "end_time")
    var endTime: Date? = null,

    @Column(name = "max_runtime")
    var maxRuntime: Int? = null,

    @Column(name = "sid")
    var sid: Int? = null,

    @Column(name = "duration", nullable = false)
    var duration: Long? = null,

    @Column(name = "tbl_comment", length = 500)
    var tblComment: String? = null,

    @Formula("LOWER(concat_ws(',', source_db || '.' || source_table , target_db || '.' || target_table , part_kind , part_name , filter))")
    val filterColumn: String? = null
)
